### ISSUE TITLE
AttachmentView's header_title is overlapping with download icon

### ISSUE THREAD

**Issue by sher999** [RocketChat/Rocket.Chat.ReactNative#6043]
### Description:

In attachmentView page viewing an image with long name, it will overlap with the right Icon (download button) 

### Environment Information:

- Rocket.Chat Server Version: 7.1.0
- Rocket.Chat App Version: 4.56.0 , 4.55.0
- Device Name: Android Poco f4, pixel 3a
- OS Version: android 14

### Steps to reproduce:

1. Open a chat on chats page 
2. then open an image on chat which has a large name
3. On the header of the attcahmentView page title and download button is overlapping 

### Expected behavior:
<details><summary> Screenshot </summary>

![Screenshot_1733328585](https://github.com/user-attachments/assets/b9158591-75c4-4301-8f03-0d4b66f2ffca)

</details>

### Actual behavior:

<details><summary>Screenshot</summary>

![Screenshot_1733328563](https://github.com/user-attachments/assets/6ed0d889-1d03-457e-80b6-992cfebb87db)

<details>
### Additional context:

tried solving it with trimming the tittle to 20 to 30 subString
```
const options = {
			title: title || '',
			headerLeft: () => (
				<HeaderButton.CloseModal testID='close-attachment-view' navigation={navigation} color={colors.fontDefault} />
			),
			
			headerRight: () => 
				Allow_Save_Media_to_Gallery && !isImageBase64(attachment.image_url) ? (
					<HeaderButton.Download testID='save-image' onPress={handleSave} color={colors.fontDefault} />
				) : null
		};
```

**Comment by Rohit3523** [comment]
This was a known issue in React Native Screens, and it was recently fixed in [React Native Screens v4](https://github.com/software-mansion/react-native-screens/releases/tag/4.0.0).

```
native header now uses flexbox model (in VDOM) - long story short - long titles are now truncated properly 
```

**Comment by sher999** [comment]
Should I continue working on it ?

**Comment by Rohit3523** [comment]
Sure, try upgrading the package first.

**Comment by sher999** [comment]
Upgraded `react-native-screens @4.3.0 `  doesn't work on android ,
<img width="881" alt="Screenshot 2024-12-05 at 1 09 13 PM" src="https://github.com/user-attachments/assets/bf2b6401-b3ed-432c-9a8e-99a1fe06d453">


 adding `headerTitleAlign` to ` center ` works neatly

<img width="881" alt="Screenshot 2024-12-05 at 1 09 01 PM" src="https://github.com/user-attachments/assets/5b2efd57-8d81-465a-8bee-5600076cbc55">
</br>

here is the screenshot of Ios and android 

![Screenshot_1733385570](https://github.com/user-attachments/assets/e92290eb-6fb1-4775-98ae-6df1df353156)
![Simulator Screenshot - iPhone 16 Pro - 2024-12-05 at 13 09 28](https://github.com/user-attachments/assets/831867aa-db01-4b17-a1a6-af51350f3a41)

**Comment by Rohit3523** [comment]
We're currently using `react-navigation/native-stack` version 6, but the fix we need is still in a draft PR ([#12135](https://github.com/react-navigation/react-navigation/pull/12135)).  

We have two options:  
1. Upgrade to version 7 ([release notes](https://reactnavigation.org/blog/2024/11/06/react-navigation-7.0/)) and see if it solves the issue.
2. Wait for the PR to be merged and released for version 6.

Can you try upgrading the package to version 7 and test the navigation flows?

**Comment by sher999** [comment]
Yeah sure trying for version 7.

**Comment by sher999** [comment]
Yes it worked with updating the `react-navigation/native-stack ` to `"7.1.10"`  and `@react-navigation/native` to `"7.0.9"`

![Screenshot_1733395192](https://github.com/user-attachments/assets/47a7c0fa-d15e-4419-b35c-deb533857412)


thanks for your guidance.

**Comment by Rohit3523** [comment]
I will have to check the changes in navigation library as it's a major release by them
